esp/services/esdl_svc_engine/esdl_binding.hpp

```diff
@@ -458,43 +455,50 @@ class EsdlBindingImpl : public CHttpSoapBinding, implements IEsdlListener
         return true;
     }
 
-/*
     virtual bool attach()
     {
-        CriticalBlock b(detachCritSec);
-        ESPLOG(LogNormal, "Binding %s.%s is subscribing to Dali for ESDL changes...", m_processName.get(), m_bindingName.get() );
-        if (m_pBindingSubscription)
+        if (isAttached())
         {
-            ESPLOG(LogNormal, "ESDL Binding %s is subscribing to all /ESDL/Bindings/Binding dali changes", this->m_bindingName.get());
-            m_pBindingSubscription->subscribe();
+            ESPLOG(LogNormal, "Binding %s.%s is already attached to Dali for ESDL changes...", m_processName.get(), m_bindingName.get() );
+            return true;
         }
 
-        if (m_pDefinitionSubscription)
+        if (m_pSubscription)
         {
-            ESPLOG(LogNormal, "ESDL Binding %s is subscribing to all /ESDL/Bindings/Definition dali changes", this->m_bindingName.get());
-            m_pDefinitionSubscription->subscribe();
+            ESPLOG(LogNormal, "Binding %s.%s is subscribing to Dali for ESDL changes...", m_processName.get(), m_bindingName.get() );
+            m_pSubscription->subscribe();
+
+            ESPLOG(LogNormal, "Requesting reload of ESDL %s.%s binding...", m_processName.get(), m_bindingName.get() );
+            reloadBindingFromCentralStore(m_bindingName.get(), m_processName.get());
+            setIsAttached(true);
+
+            /* rodrigo
```

Comment:
@mayx while rebasing your ESDLStore changes, I realized we're only subscribing to changes in /ESDL/Bindings not in /ESDL/Definitions. Was that intentional? We need to be notified of changes to the current definition.